src/common/src/metrics/guarded_metrics.rs

```diff
@@ -118,12 +135,15 @@ pub type LabelGuardedIntCounterVec<const N: usize> =
     LabelGuardedMetricVec<VecBuilderOfCounter<AtomicU64>, N>;
 pub type LabelGuardedIntGaugeVec<const N: usize> =
     LabelGuardedMetricVec<VecBuilderOfGauge<AtomicI64>, N>;
+pub type LabelGuardedUintGaugeVec<const N: usize> =
```

Comment:
Is this really necessary? Why not  use `LabelGuardedIntGaugeVec` directly?

Reply:
Because the icelake uses Uint. For semantics, I think it's correct, like `iceberg_rolling_unfushed_data_file` it will never smaller than 0.🤔

Reply:
Both are ok to me. It's just maybe we don't need to be so safe for metrics😂